omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/PersonAttributeResource1_8.java

```diff
@@ -113,6 +94,42 @@ public void setHydratedObject(PersonAttribute personAttribute, String attributab
         }
     }
 
+	@PropertySetter("value")
+	public  void setValue(PersonAttribute personAttribute, String value) {
+		PersonAttributeType attributeType = personAttribute.getAttributeType();
+		if(attributeType == null) {
+			personAttribute.setValue(value);
+		} else {
+			// Check if expected value is attributable and do the right thing.
+			try {
+				String format = personAttribute.getAttributeType().getFormat();
+				if(format == null) {
+					personAttribute.setValue(value);
+				} else {
+					Class<?> clazz = Context.loadClass(personAttribute.getAttributeType().getFormat());

```

Comment:
You could as well have used the already existing format variable instead of making the call again to get the format.